content/extending-ui/extending-chat-widget/rich-messages/index.mdx

```diff
@@ -0,0 +1,272 @@
+* `elements` - optional, this is the array of elements included in the Rich Message, for example multiple cards. There can be a maximum of 11 elements.
+* `elements.title` and `elements.subtitle` - both optional, displays the formatted text on Cards.
+* `elements.image` - optional, displays the image on the card. Can have the following optional parameters: `name`, `url`, `content_type`, `size`, `width`, `height`. 
+* `elements.buttons` - optional, array of the buttons displayed (maximum of 11). They can have the following parameters (all required if the `elements.buttons` is included):
+    * `elements.buttons.type` - defines the behavior after user clicks on the button. Might have the following values:
+        * `webview` - opens the [Moment](/extending-ui/extending-chat-widget/chat-widget-moments/),
+        * `message` - sends a message postback in the chat,
+        * `url` - opens given URL in the new tab,
+        * `phone` - allows the Customer to call the number given directly from the chat,
+    * `elements.buttons.postback_id` - description of the action sent,
+    * `elements.buttons.user_ids` - describes the user that sent the postback, can be empty,
+    * `elements.buttons.value` - defines what value should the button have, depending on its type: either a message to be sent, url to be opened in the Moment or in the new tab, or the phone number to call,
+    * `elements.buttons.webview_height` - required only if the `type` of the button is set to `webview`, defines the size of the Moment shown to the user. Might be set to `compact`, `full` or `tall`.
+
+Huge advantage of the fact that most of those parameters are optional is that it gives a lot of freedom and possibilities to unleash your creativity.
+
+# Usage examples
+Now, when the power of the Rich Messages is revealed, we can test out some of the opportunities it gives us.
+
+## Predefined answers to choose from
+Imagine that you ask your Customer the same questions over and over again, for which the answer does not require them to be especially creative. For example, if you run the support it might be "Which OS are you using?" - you would like to know whether it is a "Windows","MacOS" or "Other", so you can safely and easily guide them through the data gaining process. You can use the `quick_replies` to to that:
+
+<CodeSample>
+
+```json
+{
+	"template_id": "quick_replies",
+	"elements": [{
+            "title": "What OS are you using?",
+			"buttons": [{
+				"type": "message",
+				"text": "Windows",
+				"postback_id": "send_message",
+				"user_ids": [],
+				"value": "Windows"
+			},
+            {
+                "type": "message",
+				"text": "MacOS",
+				"postback_id": "send_message",
+				"user_ids": [],
+				"value": "MacOS"
+            },
+            {
+                "type": "message",
+				"text": "Other",
+				"postback_id": "send_message",
+				"user_ids": [],
+				"value": "Other"
```

Comment:
Linux? :pepe_grand:

Reply:
Well, it says "Other". :trollface: 

Will update next year, as probably 2020 would be the year of Linux on the Desktop.

Reply:
:trollface: